indsl/ts_utils/array_operations.py

```diff
@@ -0,0 +1,127 @@
+import numpy as np
+
+# Simple operations
+import pandas as pd
+
+from indsl.exceptions import UserValueError
+from indsl.type_check import check_types
+
+
+@check_types
+def time_weighted_mean(data: pd.Series) -> pd.Series:
+    """Time weighted mean.
+
+    The time weighted mean for a time series. The returned timeseries has a constant value
+
+    Args:
+        data: Time series.
+
+    Returns:
+        pandas.Series: Time weighted mean.
+    """
+    from scipy.integrate import trapezoid
+
+    n = len(data)
+    if n == 0:
+        raise UserValueError("Expected at least one item in data parameter, got zero instead.")
+    # We need to convert the datetime to timestamp
+    timestamps = np.array([val.timestamp() for val in data.index])
+
+    # integrate over the time series to get the sum
+    timeseries_sum = trapezoid(data.values, x=timestamps)
+    # scale by the time range for the integration, giving the time weighted average
+    timeseries_average = timeseries_sum / (timestamps[-1] - timestamps[0])
+    # We need to convert it into a pandas Series
+    time_weighted_mean = pd.Series(timeseries_average * np.ones(data.shape), index=data.index)
+
+    return time_weighted_mean
+
+
+@check_types
+def time_weighted_std(data: pd.Series) -> pd.Series:
+    """Time weighted std.
+
+    The time weighted standard deviation for a time series. The returned timeseries has a constant value
+
+
+    Args:
+        data: Time series.
+
+    Returns:
+        pandas.Series: Time weighted std.
+    """
+    from scipy.integrate import trapezoid
+
+    n = len(data)
```

Comment:
`validate_series_is_not_empty` could be use here as well and any other place where you check for `n==0`